packages/remix-server-runtime/cookies.ts

```diff
@@ -113,7 +113,9 @@ export const createCookieFactory =
         return name in cookies
           ? cookies[name] === ""
             ? ""
-            : await decodeCookieValue(unsign, cookies[name], secrets)
+            : cookies[name] !== undefined
+            ? await decodeCookieValue(unsign, cookies[name]!, secrets)
+            : null
```

Comment:
Let's make this a positive condition

```suggestion
            : cookies[name] === undefined
            ? null
            : decodeCookieValue(unsign, cookies[name], secrets)
```